### PR TITLE
[Refactor] Local Meta store support serialized and deserialized with JSON format

### PR DESCRIPTION
**Pull request by HangyuanLiu**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 42 / 197 (21.32%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|3|00.00%|[468, 469, 470]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeMgr.java|0|1|00.00%|[582]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|7|00.00%|[1421, 1424, 1426, 1427, 1428, 1432, 1522]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|0|5|00.00%|[1754, 1755, 1756, 1758, 1760]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|0|48|00.00%|[350, 351, 353, 354, 355, 356, 357, 359, 366, 367, 370, 374, 375, 376, 377, 378, 380, 383, 384, 385, 387, 389, 390, 391, 392, 393, 395, 396, 397, 398, 399, 400, 401, 404, 406, 407, 408, 409, 410, 411, 414, 415, 416, 417, 418, 419, 422, 424]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|50|02.00%|[1197, 1198, 1199, 1201, 1202, 1203, 1204, 1206, 1207, 1208, 1209, 1210, 1211, 1213, 1214, 1215, 1216, 1217, 1219, 1221, 1222, 1223, 1225, 1227, 1229, 1230, 1233, 1235, 1236, 1237, 1238, 1241, 1242, 1243, 1244, 1245, 1248, 1249, 1250, 1251, 1254, 1255, 1257, 1259, 1263, 1264, 1265, 1267, 1269]|
|:large_blue_circle:|com/starrocks/catalog/EsTable.java|1|30|03.33%|[441, 443, 444, 445, 446, 447, 448, 449, 450, 452, 453, 454, 455, 458, 459, 460, 462, 464, 466, 467, 468, 469, 471, 472, 474, 476, 477, 479, 481]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|26|39|66.67%|[4707, 4708, 4721, 4723, 4724, 4725, 4726, 4727, 4739, 4751, 4752, 4754, 4755]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/FileTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|2|2|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationMgr.java [hunk 1]
```diff
@@ -740,8 +740,6 @@ public void loadV2(DataInputStream dis) throws IOException, SRMetaBlockException
                 }
             } catch (AuthenticationException e) {
                 throw new RuntimeException(e);
-            } finally {
-                reader.close();
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationMgr.java:744]: why remove it? miss checksum checking?

File: fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java [hunk 2]
```diff
@@ -197,15 +201,8 @@ public synchronized boolean recyclePartition(long dbId, long tableId,
...
-        RecyclePartitionInfo partitionInfo = null;
-        if (isCloudNativeTable) {
-            // lake table
-            partitionInfo = new RecycleRangePartitionInfo(dbId, tableId, partition,
-                    range, dataProperty, replicationNum, isInMemory, storageCacheInfo);
-        } else {
-            partitionInfo = new RecyclePartitionInfoV1(dbId, tableId, partition,
-                    range, dataProperty, replicationNum, isInMemory);
-        }
+        RecyclePartitionInfo partitionInfo = new RecycleRangePartitionInfo(dbId, tableId, partition,
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/CatalogRecycleBin.java:204]: @sduzh plz help to check whether we can merge these two together

File: fe/fe-core/src/main/java/com/starrocks/catalog/EsTable.java [hunk 3]
```diff
@@ -86,6 +88,15 @@ public class EsTable extends Table {
...
     public static final String KEY_TIME_ZONE = "time_zone";
 
+    // tableContext is used for being convenient to persist some configuration parameters uniformly
+    @SerializedName(value = "tc")
+    private Map<String, String> tableContext = new HashMap<>();
+
+    // only save the partition definition, save the partition key,
+    // partition list is got from es cluster dynamically and is saved in esTableState
+    @SerializedName(value = "p")
+    private PartitionInfo partitionInfo;
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/EsTable.java:98]: need to register PartitionInfo and its subclasses in GSONUtils?

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java [hunk 4]
```diff
@@ -94,16 +95,20 @@ public enum HudiTableType {
...
+    @SerializedName(value = "rn")
+    private String resourceName;
+    @SerializedName(value = "pcn")
     private List<String> partColumnNames = Lists.newArrayList();
     // dataColumnNames stores all the non-partition columns of the hudi table,
     // consistent with the order defined in the hudi table
+    @SerializedName(value = "dcn")
     private List<String> dataColumnNames = Lists.newArrayList();
-
+    @SerializedName(value = "prop")
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:111]: need to register Table and all its subclasses?

File: fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationMgr.java (hunk 1 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/authentication/AuthenticationMgr.java:744]: The above close has been verified, here is the second verification, this is a bug in the previous PR

File: fe/fe-core/src/main/java/com/starrocks/qe/VariableMgr.java [hunk 5]
```diff
@@ -341,6 +346,83 @@ public static void read(DataInputStream in) throws IOException, DdlException {
...
+                    continue;
+                }
+                switch (field.getType().getSimpleName()) {
+                    case "boolean":
+                    case "int":
+                    case "long":
+                    case "float":
+                    case "double":
+                    case "String":
+                        m.put(attr.name(), field.get(DEFAULT_SESSION_VARIABLE).toString());
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/qe/VariableMgr.java:366]: do we need to filter the `SESSION_ONLY` variables here?

File: fe/fe-core/src/main/java/com/starrocks/catalog/EsTable.java (hunk 3 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/catalog/EsTable.java:98]: PartitionInfo has register subtype in GSONUtils

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 57 / 226 (25.22%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|3|00.00%|[468, 469, 470]|
|:large_blue_circle:|com/starrocks/statistic/AnalyzeMgr.java|0|1|00.00%|[582]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|8|00.00%|[1430, 1433, 1435, 1436, 1437, 1439, 1441, 1533]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|0|5|00.00%|[1754, 1755, 1756, 1758, 1760]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|0|51|00.00%|[351, 352, 354, 355, 356, 357, 358, 360, 367, 368, 371, 375, 376, 377, 378, 379, 381, 388, 389, 392, 396, 397, 398, 400, 402, 403, 404, 405, 407, 408, 409, 410, 411, 412, 415, 417, 418, 419, 420, 421, 424, 425, 426, 427, 428, 431, 433, 756, 757, 758, 759]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|50|02.00%|[1197, 1198, 1199, 1201, 1202, 1203, 1204, 1206, 1207, 1208, 1209, 1210, 1211, 1213, 1214, 1215, 1216, 1217, 1219, 1221, 1222, 1223, 1225, 1227, 1229, 1230, 1233, 1235, 1236, 1237, 1238, 1241, 1242, 1243, 1244, 1245, 1248, 1249, 1250, 1251, 1254, 1255, 1257, 1259, 1263, 1264, 1265, 1267, 1269]|
|:large_blue_circle:|com/starrocks/catalog/EsTable.java|1|30|03.33%|[441, 443, 444, 445, 446, 447, 448, 449, 450, 452, 453, 454, 455, 458, 459, 460, 462, 464, 466, 467, 468, 469, 471, 472, 474, 476, 477, 479, 481]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|28|51|54.90%|[4703, 4704, 4711, 4712, 4713, 4714, 4715, 4716, 4729, 4730, 4731, 4732, 4733, 4736, 4737, 4738, 4739, 4740, 4752, 4764, 4765, 4767, 4768]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/FileTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ExpressionRangePartitionInfo.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java (hunk 4 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/catalog/HudiTable.java:111]: done

File: fe/fe-core/src/main/java/com/starrocks/qe/VariableMgr.java (hunk 5 above)
Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/qe/VariableMgr.java:367]: done

**Review by gengjun-git — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 64 / 232 (27.59%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/Table.java|0|3|00.00%|[468, 469, 470]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|0|8|00.00%|[1430, 1433, 1435, 1436, 1437, 1439, 1441, 1533]|
|:large_blue_circle:|com/starrocks/privilege/AuthorizationMgr.java|0|5|00.00%|[1755, 1756, 1757, 1759, 1761]|
|:large_blue_circle:|com/starrocks/qe/VariableMgr.java|0|51|00.00%|[352, 353, 355, 356, 357, 358, 359, 361, 368, 369, 372, 376, 377, 378, 379, 380, 382, 389, 390, 393, 397, 398, 399, 401, 403, 404, 405, 406, 408, 409, 410, 411, 412, 413, 416, 418, 419, 420, 421, 422, 425, 426, 427, 428, 429, 432, 434, 757, 758, 759, 760]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|1|50|02.00%|[1198, 1199, 1200, 1202, 1203, 1204, 1205, 1207, 1208, 1209, 1210, 1211, 1212, 1214, 1215, 1216, 1217, 1218, 1220, 1222, 1223, 1224, 1226, 1228, 1230, 1231, 1234, 1236, 1237, 1238, 1239, 1242, 1243, 1244, 1245, 1246, 1249, 1250, 1251, 1252, 1255, 1256, 1258, 1260, 1264, 1265, 1266, 1268, 1270]|
|:large_blue_circle:|com/starrocks/catalog/EsTable.java|1|30|03.33%|[441, 443, 444, 445, 446, 447, 448, 449, 450, 452, 453, 454, 455, 458, 459, 460, 462, 464, 466, 467, 468, 469, 471, 472, 474, 476, 477, 479, 481]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|28|51|54.90%|[4704, 4705, 4712, 4713, 4714, 4715, 4716, 4717, 4730, 4731, 4732, 4733, 4734, 4737, 4738, 4739, 4740, 4741, 4753, 4765, 4766, 4768, 4769]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/RangePartitionInfo.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/FileTable.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/gson/GsonUtils.java|14|14|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/persist/metablock/SRMetaBlockID.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/catalog/ExpressionRangePartitionInfo.java|1|1|100.00%|[]|

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**